src/gt4py/cartesian/gtc/passes/oir_optimizations/temporaries.py

```diff
@@ -25,7 +25,9 @@ def visit_FieldAccess(
         offsets = node.offset.to_dict()
         if node.name in tmps_name_map:
             assert offsets["i"] == offsets["j"] == offsets["k"] == 0, (
-                "Non-zero offset in temporary that is replaced?!"
+                "No K-offset capabilities of 3D temporaries. Must define '"
```

Comment:
The check up top checks for `i` and `j` to also be offset. Why does the error message change to k only?

And you have the option to offset temporaries in k: 
```python 
    with computation(PARALLEL), interval(...):
        tmp = 100
    with computation(PARALLEL), interval(1, None):
        out_field = tmp[0, 0, -1]
```
is legit gtp4y code